Signal/src/ViewControllers/HomeViewController.m

```diff
@@ -111,7 +110,7 @@ - (void)commonInit
     _blockingManager = [OWSBlockingManager sharedManager];
     _blockedPhoneNumberSet = [NSSet setWithArray:[_blockingManager blockedPhoneNumbers]];
 
-    _experienceUpgradeFinder = [ExperienceUpgradeFinder new];
+    ExperienceUpgradeFinder.sharedManager;
```

Comment:
This reads like a no-op. Why are we calling the sharedManager here and not doing anything with it? Can you either delete this line or leave a comment about why this line is here.

TIOLI: Rather than sprinkling references to singletons all over the code base, hiding it behind either a getter or an assignable property makes it easier for future dependency stubbing.

e.g. 

either: 

```
_experienceUpgradeFinder =  ExperienceUpgradeFinder.sharedManager
```

or 

```
- (ExperienceUpgradeFinder *)experienceUpgradeFinder
{
    return ExperienceUpgradeFinder.sharedManager;
}
```

Reply:
What do you think of my proposal for a singleton registry?  It's the best way I know of to safely and easily manage singletons...

As for why this is here, for now I'm just trying to make sure that singletons are still constructed in the same order they used to be.  In the long run, I'd like to do something quite different.